### PR TITLE
Optionally register in JS

### PR DESCRIPTION
**Pull request by psealock**
Fixes https://github.com/woocommerce/navigation/issues/111

Avoiding a double registration in PHP and JS seems inevitable (See p90Yrv-1TY) so this PR explores only needing to provide a JS Fill if desired.

Doing so frees up WC Navigation to assume Menus are always handled by PHP, so `WooNavigationMenu` can be removed altogether, leaving `WooNavigationItem` as the only thing required to render a custom link.

Slots and Fills generate a `name` made of `item.parentMenu` and `item.id`, so a unique Slot exists for a `WooNavigationMenu` matching those parameters of location in the hierarchy.

### New API 

Just like before, plugins need to register in PHP.

```php
function register_items() {
	if ( method_exists( '\Automattic\WooCommerce\Navigation\Menu', 'add_category' ) ) {
		\Automattic\WooCommerce\Navigation\Menu::add_category(
			array(
				'id'         => 'examples-root',
				'title'      => 'Examples',
				'capability' => 'view_woocommerce_reports',
			)
		);
	}

	if ( method_exists( '\Automattic\WooCommerce\Navigation\Menu', 'add_item' ) ) {
		\Automattic\WooCommerce\Navigation\Menu::add_item(
			array(
				'id'         => 'example-1',
				'parent'     => 'examples-root',
				'title'      => 'Example 1',
				'capability' => 'view_woocommerce_reports',
				'url'        => 'http//:www.google.com',
			)
		);
	}

	if ( method_exists( '\Automattic\WooCommerce\Navigation\Menu', 'add_item' ) ) {
		\Automattic\WooCommerce\Navigation\Menu::add_item(
			array(
				'id'         => 'example-2',
				'parent'     => 'examples-root',
				'title'      => 'Example 2',
				'capability' => 'view_woocommerce_reports',
				'url'        => 'http//:www.google.com',
			)
		);
	}
}

add_action( 'plugins_loaded', 'register_items' );
```

The menu works as it should.

Screen Shot 2020-09-28 at 5 07 31 PM

Now, plugins only need to render an array of `WooNavigationItem` for nav items they'd like to customize.

```js
const { WooNavigationItem } = window.wc;

const MyPlugin = () => {
  return (
    <WooNavigationItem parentMenu="examples-root" item="example-1">
      <a className="components-button" href="www.example.com/1">
        Example 1 😜
      </a>
    </WooNavigationItem>
  );
};

registerPlugin("my-plugin", { render: MyPlugin });
```
A Fill is detected for the nav item and a slot is rendered.

Screen Shot 2020-09-28 at 5 08 33 PM

### Test Instructions

1. In your plugins folder clone, install, and start https://github.com/psealock/examples. Or, if you already have it, be sure to `git pull origin master`.
2. Link `@wordpress/components` in this repository, if you haven't already.
3. See the Example links functioning in the sidebar.

### Improvements

The id creation of `category.id + item.id` can be simplified. As it is, `WooNavigationItem` requires a correct `parentMenu` and `item`, but maybe all that is required is the `item`.

### CODE REVIEW

File: client/components/navigation-container/index.js [hunk 1]
```diff
@@ -18,10 +19,26 @@ import { withSelect } from '@wordpress/data';
  */
 import { NAVIGATION_STORE_NAME } from '../../store';
 import NavigationHeader from '../navigation-header';
-import {
-	WooNavigationItem,
-	WooNavigationMenu,
-} from '../../../packages/components/woo-navigation';
+import { WooNavigationItem } from '../../../packages/components/woo-navigation';
+
+const NavigationContainerItem = ( { category, item } ) => {
```
Comment (joshuatf) [inline client/components/navigation-container/index.js:24]: Just a heads up that this name should be updated to follow the changes in https://github.com/woocommerce/navigation/pull/116

**Review by joshuatf — COMMENTED** [review]
Really love the balance you found here between JS and PHP.  This seems like the best path forward for now. ❤️ 

>The id creation of category.id + item.id can be simplified. As it is, WooNavigationItem requires a correct parentMenu and item, but maybe all that is required is the item.

I like the simplicity of just using the item ID.

I'm also not seeing the updates in the examples repo and hitting some errors in React.  Is there another commit after "smiley?"

**Comment by psealock** [comment]
I clean up the slotFill name to be mapped to `item.id` instead of item and category ids and introduced a new internal compnent `Item`.

**Review by joshuatf — APPROVED** [review]
Looks great and testing well! 🎉 🌮  LGTM

**Pull request merged**